### ISSUE TITLE
Is there an option for the process to return non-zero code if there are any warnings?

### ISSUE THREAD

**Issue by dusan-milosevic-the-real-one** [kawa1214/import-lint#24]
I integrated lint checks in the pipeline, ad I would like it to fail if there are any warnings.

For other linters, there's usually an option like `--fatal-warnings` that would make the process (and the step in the pipeline) to fail if there are any warnings, but it looks like this plugin ignores the command line args that were passed. Is there a way to achieve this?

**Comment by kawa1214** [comment]
@dusan-milosevic-the-real-one 

I would like to implement that feature, but currently do not have it.

For example, how about this shell script?
I think it fails with an error code of 1 if there is a warning.

```sh
import_lint=$(flutter pub run import_lint)
if [ ! "`echo $import_lint | grep 'No issues found'`" ]; then
    # notify
    exit 1
fi
```

**Comment by dusan-milosevic-the-real-one** [comment]
Key @kawa1214 !

Thanks for the quick response! 

It would have to be something like:

```
import_lint=$(flutter pub run import_lint)
if [  "`echo $import_lint | grep 'WARNING'`" ]; then
    # notify
    exit 1
fi
```

because the command outputs '0 issues found' and it contains 'WARNING' if there is a warning.

**Comment by kawa1214** [comment]
@dusan-milosevic-the-real-one 
Thanks. I think it's good!
Mine was a not of not.

**Comment by dominicmh** [comment]
This solution doesn't print the output of import_lint. import_lint would need to be run twice in order to get the output and the exit code.

Instead of `grep`, `awk` could be used to print entire import_lint output while exiting with corresponding exit code:
`flutter pub run import_lint | awk -v lintquery="WARNING" '$0 ~ lintquery { exitcode = 1 } 1; END { exit(exitcode) }'`

(based on [this answer](https://unix.stackexchange.com/a/307578))
